### PR TITLE
Support FakeItEasy 5.0.0+

### PR DESCRIPTION
**Pull request by blairconrad**
Moves FakeItEasy version restrictions up to `, 6.0.0)` and adds AutoFakeItEasy.FakeItEasy5UnitTest to ensure the integration works.

### CODE REVIEW

**Comment by blairconrad** [comment]
Will update the dependency in AutoFakeItEasy.FakeItEasy5UnitTest to full 5.0.0 once that's released.

**Comment by blairconrad** [comment]
And if #1090 is merged, I'd also add a .netcoreapp2.0 flavour to the tests.

File: Src/AutoFakeItEasy.FakeItEasy5UnitTest/AutoFakeItEasy.FakeItEasy5UnitTest.csproj [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  <PropertyGroup>
+    <TargetFrameworks>net452;netcoreapp1.1</TargetFrameworks>
+    <AssemblyTitle>AutoFakeItEasy.FakeItEasy5.UnitTest</AssemblyTitle>
+    <AssemblyName>AutoFixture.AutoFakeItEasy.FakeItEasy5UnitTest</AssemblyName>
+    <RootNamespace>AutoFixture.AutoFakeItEasy.UnitTest</RootNamespace>
+
+    <!--  Suppress warning about invalid dependency version in Castle.Core.
+          That is FakeItEasy dependency and we cannot fix that somehow. -->
+    <NoWarn>$(NoWarn);NU1603</NoWarn>
```
Comment (zvirja) [inline Src/AutoFakeItEasy.FakeItEasy5UnitTest/AutoFakeItEasy.FakeItEasy5UnitTest.csproj]: Is this warning this actual?

**Comment by zvirja** [comment]
@blairconrad Thanks for the follow up and making us a part of your release flow 😄 

Let's go ahead with #1090 and test `netcoreapp2.0` as well 👍

File: Src/AutoFakeItEasy.FakeItEasy5UnitTest/AutoFakeItEasy.FakeItEasy5UnitTest.csproj (hunk 1 above)
Comment (blairconrad) [reply Src/AutoFakeItEasy.FakeItEasy5UnitTest/AutoFakeItEasy.FakeItEasy5UnitTest.csproj]: Well spotted! It's still needed for AutoFakeItEasy.FakeItEasy4UnitTest, but not AutoFakeItEasy.FakeItEasy5UnitTest.

**Comment by blairconrad** [comment]
> Thanks for the follow up and making us a part of your release flow

Of course. Nobody wants to break downstream dependencies, and these glue libraries are a weird case - often contributed by someone who's not a core member of either library's team, so maintenance can be a burden.

'sides, it's good press for FakeItEasy. :wink:

Also, it was lucky. The tests in this project detected a bug I'd introduced in 5.0.0-alpha.1. https://github.com/FakeItEasy/FakeItEasy/issues/1548, in case you didn't see it.
The test isn't _intentionally_ looking for the functionality that wasn't working, but because of how it's written, it triggered the bad behaviour.

**Comment by blairconrad** [comment]
Updated to use FakeItEasy 5.0.0 in the tests.

**Comment by zvirja** [comment]
@blairconrad Merged and will be published soon! Thanks for this follow up!

**Comment by blairconrad** [comment]
> Target .NET Core 2.0 in tests
> @zvirja committed 18 hours ago

How embarrassing. Thank you. And thanks for the merge!

**Pull request merged**